src/js_native_api_v8.cc

```diff
@@ -2569,6 +2569,38 @@ napi_status napi_create_arraybuffer(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+napi_status napi_is_detached_arraybuffer(napi_env env,
+                                         napi_value value,
+                                         napi_value* result) {
+  NAPI_PREAMBLE(env);
+  CHECK_ARG(env, result);
+  v8::Isolate* isolate = env->isolate;
+
+  v8::Local<v8::Value> val = v8impl::V8LocalValueFromJsValue(value);
+
+  if (!val->IsObject()) {
+    napi_throw_type_error(env,
```

Comment:
No JavaScript exceptions were expected on N-API checks. See discussion at https://github.com/nodejs/node/issues/29849.

Reply:
You could just remove this check and let the below one handle returning false.